hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/StorageType.java

```diff
@@ -34,28 +34,35 @@
 @InterfaceStability.Unstable
 public enum StorageType {
   // sorted by the speed of the storage types, from fast to slow
-  RAM_DISK(true),
-  SSD(false),
-  DISK(false),
-  ARCHIVE(false),
-  PROVIDED(false);
+  RAM_DISK(true, true),
+  NVDIMM(false, true),
+  SSD(false, false),
+  DISK(false, false),
+  ARCHIVE(false, false),
+  PROVIDED(false, false);
 
   private final boolean isTransient;
+  private final boolean isRAM;
 
   public static final StorageType DEFAULT = DISK;
 
   public static final StorageType[] EMPTY_ARRAY = {};
 
   private static final StorageType[] VALUES = values();
 
-  StorageType(boolean isTransient) {
+  StorageType(boolean isTransient, boolean isRAM) {
     this.isTransient = isTransient;
+    this.isRAM = isRAM;
   }
 
   public boolean isTransient() {
     return isTransient;
   }
 
+  public boolean isRAM() {
+    return isRAM;
+  }
```

Comment:
will RAM_DISK and NVDIMM co-exist..? if co-exist's, why can't we name NVDIM itself..?

Reply:
> why can't we name NVDIM itself..?

I don't think I get the question totally. Yes they can co-exist. You can have some folders being RAM_DISK storage and some folders being NVDIMM storage. RAM_DISK will get data lost after process restart, so it is transient. NVDIMM is not transient.

Reply:
As this will be co-exists. I was suggesting to change the method name to something like "isNvdimm" which was revlant here

Reply:
No. Both RAM_DISK and NVDIMM will return true here.

Reply:
Balancer and mover will not move the blocks based on the `isTransient` ( they call getMovableTypes(..))..the blocks which NVDIMM shouldn't moved I feel,but as per this change it will move.

Reply:
Oh, I was thinking that allowing Balancer to move the NVDIMM data is by design since they are not volatile. But if that is case, then we can update Balancer code by replacing `isTransient()` call with `isRAM()` call. Not sure if this makes more sense?

Reply:
NVDIMM is RAM, however, the data above can be stored persistently. It can be regarded as a general hardware device. We don't have to consider what storage type it is. Therefore, I think it is better to use isRAM to determine whether to use mover . In addition, neither RAM nor nvdimm need FsDatasetCache, and isTransient() used to determine whether FsDatasetCache is needed

Reply:
> neither RAM nor nvdimm need FsDatasetCache, and isTransient() used to determine whether FsDatasetCache is needed

I think we have agreed on this.

>  I think it is better to use isRAM to determine whether to use mover 

So just to be clear, if we use `isRAM()` to determine, both RAM_DISK and NVDIMM will return true and thus will simply disable Balancer and Mover. I was proposing that only when we do not want to move those NVDIMM data replicas around (disk/node) ever. Is that the design here? I'm fine with that but I think this contradicts with the statement that _"It can be regarded as a general hardware device. "_

Reply:
Sorry, I made a big mistake in the above reply. Our design idea is: NVDIMM supports mover and balancer, and `isTransient() ` applied in the case. isRAM() is only used to “FsDatasetCache” judgment. So the current code is reasonable, i think it‘s not necessary to modify the code.

Reply:
Thanks for clarification, @YaYun-Wang I think now we both are on the same page. @brahmareddybattula Does this make sense to you? Thanks

Reply:
@brahmareddybattula  would you please have a look for this? Thanks.

Reply:
ok, By design if you dn't want to move.

Reply:
My final query then, why can't have one NVDIMM like one SSD as this also movable and peristent..?

Reply:
> My final query then, why can't have one NVDIMM like one SSD as this also movable and peristent..?


Considering NVDIMM is faster, so NVDIMM does not  use `FsDatasetCache()` which SSD needs in the design.

Reply:
ok